packages/flutter/lib/src/widgets/async.dart

```diff
@@ -220,17 +239,35 @@ class AsyncSnapshot<T> {
   /// If the asynchronous computation has never returned a value, this may be
   /// set to an initial data value specified by the relevant widget. See
   /// [FutureBuilder.initialData] and [StreamBuilder.initialData].
-  final T data;
+  T get data {
```

Comment:
The doc comment above needs updating to reflect the changed behavior.

Reply:
(And it might be worth to describe the general behavior in the doc comment of the class as well)